### ISSUE TITLE
pvc stuck in terminating state

### ISSUE THREAD

**Issue by SkinyMonkey** [rook/rook#2746]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Block storage PVC get stuck in terminating state after being attached to a completed/failed job and explicitely deleted.
removing the job/pod to which it's attached unblock the situation immediately though.

**Expected behavior:**
Block storage PVC should be deleted as soon as possible after the kubelet receive the order to remove it.

**How to reproduce it (minimal and precise):**
(btw this guide could be useful for AKS users)
- Create an AKS cluster with autocluster preview feature.
- Attach a 500Gb disk to the vm scale set in the resource group of the cluster
- Update the vmss instances with ```az vmss update-instances``` so that the disk devices appear on the nodes of the cluster.
- setup the deviceFilter field of your cluster.yaml file to "^sd[c-z]"
- setup the FLEXVOLUME_DIR_PATH value like this in your operator.yaml file:
       ```- name: FLEXVOLUME_DIR_PATH
          value: "/etc/kubernetes/volumeplugins"```
- install rook operator, cluster and a block storage
- create a job with a ceph block storage volume attached to it, wait for it to finish
- debug

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 16.04.5 LTS
* Kernel (e.g. `uname -a`):  4.15.0-1039-azure
* Cloud provider or hardware configuration: azure
* Rook version (use `rook version` inside of a Rook Pod):  v0.9.0-208.g8e1bc5b
* Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:48:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.5", GitCommit:"51dd616cdd25d6ee22c83a858773b607328a18ec", GitTreeState:"clean", BuildDate:"2019-01-16T18:14:49Z", GoVersion:"go1.10.7", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):  azure AKS
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox] (https://rook.io/docs/Rook/master/toolbox.html)): ceph 
```  cluster:
    id:     bab6e3bc-979a-46da-8183-5adbcbe6c460
    health: HEALTH_OK

  services:
    mon: 3 daemons, quorum a,c,b
    mgr: a(active)
    osd: 6 osds: 6 up, 6 in

  data:
    pools:   1 pools, 256 pgs
    objects: 308  objects, 290 MiB
    usage:   33 GiB used, 1.5 TiB / 1.5 TiB avail
    pgs:     256 active+clean
```

I would be happy to give more details as needed.

you'll find some here (extract of agent, operator and kubelet logs) : https://pastebin.com/qewP0XmJ

**Comment by SkinyMonkey** [comment]
could be related to https://github.com/kubernetes/kubernetes/issues/69697 ?

**Comment by SkinyMonkey** [comment]
```
Mar 01 17:12:45 aks-nodepool1-34421062-vmss000002 kubelet[1781]: E0301 17:12:45.937515    1781 plugin_watcher.go:121] error could not find plugin for deleted file /var/lib/kubelet/plugins/ceph.rook.io/rook-ceph-system/mounts/pvc-2c0cf7b1-3c45-11e9-b109-06d2222521f6
```
when handling delete event: "/var/lib/kubelet/plugins/ceph.rook.io/rook-ceph-system/mounts/pvc-2c0cf7b1-3c45-11e9-b109-06d2222521f6": REMOVE

**Comment by ravibagri5** [comment]
You can get rid of this issue by manually editing the pv and then removing the finalizers which looked something like this:

`- kubernetes.io/pv-protection
`
e.g

```
kubectl patch pvc pvc_name -p '{"metadata":{"finalizers":null}}'
kubectl patch pv pv_name -p '{"metadata":{"finalizers":null}}'
kubectl patch pod pod_name -p '{"metadata":{"finalizers":null}}'
```

**Comment by theonewolf** [comment]
@ravibagri5 isn't that a bug if it **requires manual intervention**?

**Comment by mukul-c2fo** [comment]
@theonewolf I think it's a safety feature to prevent the deletion of PVCs which are being actively used by a pod. More context is given in the documentation: 

https://kubernetes.io/docs/concepts/storage/persistent-volumes/

**Comment by samceena** [comment]
removing or setting the finalizers to null works.

**Comment by computerquip-work** [comment]
Should I start a new issue up? This still seems to be a problem, regardless of volume attachment being present or not.
